storage/fanout.go

```diff
@@ -262,6 +262,7 @@ func (q *mergeQuerier) SelectSorted(params *SelectParams, matchers ...*labels.Ma
 			} else {
 				priErr = qryResult.selectError
 			}
+			continue
 		}
 		seriesSets = append(seriesSets, qryResult.set)
 	}
```

Comment:
What would you think on inlining

```
		if qryResult.selectError != nil {
			q.failedQueriers[qryResult.qr] = struct{}{}
			// If the error source isn't the primary querier, return the error as a warning and continue.
			if qryResult.qr != q.primaryQuerier {
				warnings = append(warnings, qryResult.selectError)
			} else {
				priErr = qryResult.selectError
			}
			continue
		}
		seriesSets = append(seriesSets, qryResult.set)
```

to

```
		if qryResult.selectError == nil {
			seriesSets = append(seriesSets, qryResult.set)
			continue
                }
		q.failedQueriers[qryResult.qr] = struct{}{}
		// If the error source isn't the primary querier, return the error as a warning and continue.
		if qryResult.qr != q.primaryQuerier {
			warnings = append(warnings, qryResult.selectError)
		} else {
			priErr = qryResult.selectError
		}
```

Reply:
Go style suggests handling the error case first, and leaving the normal case un-indented. I don’t feel super strongly but would prefer to stay with that idiom.

Reply:
ok!

Reply:
I like this suggestion.

Reply:
The [Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow) specifies trying to keep the normal path at minimal indentation, and handling errors first. If you still think I should change it I am happy to though! 

In this case I can see either way, I kind of like that we handle warnings and errors together, but the error handling is complicated enough that I understand trying to un-indent it.

Reply:
I.e. the suggested flow ends up with:
```
handle warnings
handle success case
handle error case (possibly by adding a warning)
```
which is a bit strange.

Reply:
You are correct. I did not take context into account.

Reply:
@bwplotka Do you agree with merging the PR as it? What @csmarchbanks makes sense, too.